src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+import React, { Component } from 'react';
+
+import { TelemetryService } from 'services';
+import { DeviceIcon } from './deviceIcon';
+import { RulesGrid, rulesColumnDefs } from 'components/pages/rules/rulesGrid';
+import { translateColumnDefs } from 'utilities';
+import {
+  Flyout,
+  FlyoutHeader,
+  FlyoutTitle,
+  FlyoutCloseBtn,
+  FlyoutContent,
+  FlyoutSection
+} from 'components/shared';
+import {
+  PropertyGrid as Grid,
+  PropertyRow as Row,
+  PropertyHeaderCell as HeaderCell,
+  PropertyCell as Cell
+} from './propertyGrid';
+
+import './deviceDetails.css';
+
+export class DeviceDetails extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      alarms: undefined,
+      isAlarmsPending: false,
+      alarmsError: undefined
+    };
+    this.columnDefs = [
+      rulesColumnDefs.ruleName,
+      rulesColumnDefs.severity,
+      rulesColumnDefs.explore
+    ];
+  }
+
+  componentDidMount() {
+    if (!this.props.rulesLastUpdated) this.props.fetchRules();
+    this.fetchAlarms((this.props.device || {}).id);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if ((this.props.device || {}).id === nextProps.device.id) {
+      this.fetchAlarms((nextProps.device || {}).id);
+    }
+  }
+
+  componentWillUnmount() {
+    this.getAlarmsObsvStream.unsubscribe();
+  }
+
+  applyRuleNames = (alarms, rules) =>
+    alarms.map(alarm => ({
+      ...alarm,
+      name: (rules[alarm.ruleId] || {}).name
+    }));
+
+  fetchAlarms = (deviceId) => {
+    this.setState({ isAlarmsPending: true });
+    const getAlarmsObsv =
+      TelemetryService.getAlarms({
+        limit: 5,
+        order: "desc",
+        devices: deviceId
+      });
+
+    this.getAlarmsObsvStream = getAlarmsObsv
+      .subscribe(
+      alarms => this.setState({ alarms, isAlarmsPending: false, alarmsError: undefined }),
```

Comment:
indentation

Reply:
Sure, but let's check your settings against mine... because I can change the indentation manually... but the automatic is just going to change it back.